when allowAutoTopicCreation=false , resource group limit will fail to start

**Describe the bug**
A clear and concise description of what the bug is.
https://github.com/apache/pulsar/wiki/PIP-82%3A-Tenant-and-namespace-level-rate-limiting

when allowAutoTopicCreation=false , resource group  limit will fail to start

**To Reproduce**
Steps to reproduce the behavior:

add

```
resourceUsageTransportClassName=org.apache.pulsar.broker.resourcegroup.ResourceUsageTopicTransportManager

allowAutoTopicCreation=false
```


then start broker

```
22:37:17.036 [main] ERROR org.apache.pulsar.PulsarStandaloneStarter - Failed to start pulsar service.
org.apache.pulsar.broker.PulsarServerException: java.lang.reflect.InvocationTargetException
	at org.apache.pulsar.broker.PulsarService.start(PulsarService.java:795) ~[classes/:?]
	at org.apache.pulsar.PulsarStandalone.start(PulsarStandalone.java:301) ~[classes/:?]
	at org.apache.pulsar.PulsarStandaloneStarter.main(PulsarStandaloneStarter.java:131) [classes/:?]
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_322]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:1.8.0_322]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_322]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_322]
	at org.apache.pulsar.broker.PulsarService.start(PulsarService.java:775) ~[classes/:?]
	... 2 more
Caused by: org.apache.pulsar.client.api.PulsarClientException$TopicDoesNotExistException: {"errorMsg":"Topic non-persistent://pulsar/system/resource-usage does not exist","reqId":2195930858616055387, "remote":"localhost/127.0.0.1:6650", "local":"/127.0.0.1:51894"}
	at org.apache.pulsar.client.api.PulsarClientException.unwrap(PulsarClientException.java:1066) ~[classes/:?]
	at org.apache.pulsar.client.impl.ReaderBuilderImpl.create(ReaderBuilderImpl.java:77) ~[classes/:?]
	at org.apache.pulsar.broker.resourcegroup.ResourceUsageTopicTransportManager$ResourceUsageReader.<init>(ResourceUsageTopicTransportManager.java:129) ~[classes/:?]
	at org.apache.pulsar.broker.resourcegroup.ResourceUsageTopicTransportManager.<init>(ResourceUsageTopicTransportManager.java:223) ~[classes/:?]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_322]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:1.8.0_322]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_322]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_322]
	at org.apache.pulsar.broker.PulsarService.start(PulsarService.java:775) ~[classes/:?]
```

starting broker needs this non-persistent topic exists, but only this broker started ,we can create the non-persistent topic..

dead lock..

**Expected behavior**

allow users set the topic name, they can use  a  persistent topic

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Desktop (please complete the following information):**
 - OS: [e.g. iOS]

**Additional context**
Add any other context about the problem here.

--- a/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java
+++ b/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java
@@ -2836,7 +2836,8 @@ public class BrokerService implements Closeable {
 
     public boolean isAllowAutoTopicCreation(final TopicName topicName) {
         //System topic can always be created automatically
-        if (pulsar.getConfiguration().isSystemTopicEnabled() && checkTopicIsEventsNames(topicName)) {
+        if ((pulsar.getConfiguration().isSystemTopicEnabled() && checkTopicIsEventsNames(topicName))
+                || NamespaceName.SYSTEM_NAMESPACE.equals(topicName.getNamespaceObject())) {
             return true;
         }
         AutoTopicCreationOverride autoTopicCreationOverride = getAutoTopicCreationOverride(topicName);

--- a/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java
+++ b/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java
@@ -2835,7 +2835,8 @@ public class BrokerService implements Closeable {
 
     public boolean isAllowAutoTopicCreation(final TopicName topicName) {
         //System topic can always be created automatically
-        if (isSystemTopic(topicName)) {
+        if ((pulsar.getConfiguration().isSystemTopicEnabled() && isSystemTopic(topicName))
+            || NamespaceName.SYSTEM_NAMESPACE.equals(topicName.getNamespaceObject())) {
             return true;
         }
         AutoTopicCreationOverride autoTopicCreationOverride = getAutoTopicCreationOverride(topicName);

--- a/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java
+++ b/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java
@@ -24,7 +24,6 @@ import static org.apache.bookkeeper.mledger.util.SafeRun.safeRun;
 import static org.apache.commons.collections.CollectionUtils.isEmpty;
 import static org.apache.commons.lang3.StringUtils.isNotBlank;
 import static org.apache.pulsar.broker.PulsarService.isTransactionSystemTopic;
-import static org.apache.pulsar.common.events.EventsTopicNames.checkTopicIsEventsNames;
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.Lists;
@@ -2836,8 +2835,7 @@ public class BrokerService implements Closeable {
 
     public boolean isAllowAutoTopicCreation(final TopicName topicName) {
         //System topic can always be created automatically
-        if ((pulsar.getConfiguration().isSystemTopicEnabled() && checkTopicIsEventsNames(topicName))
-                || NamespaceName.SYSTEM_NAMESPACE.equals(topicName.getNamespaceObject())) {
+        if (isSystemTopic(topicName)) {
             return true;
         }
         AutoTopicCreationOverride autoTopicCreationOverride = getAutoTopicCreationOverride(topicName);

--- a/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java
+++ b/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java
@@ -2835,7 +2835,7 @@ public class BrokerService implements Closeable {
 
     public boolean isAllowAutoTopicCreation(final TopicName topicName) {
         //System topic can always be created automatically
-        if ((pulsar.getConfiguration().isSystemTopicEnabled() && isSystemTopic(topicName))) {
+        if (pulsar.getConfiguration().isSystemTopicEnabled() && isSystemTopic(topicName)) {
             return true;
         }
         AutoTopicCreationOverride autoTopicCreationOverride = getAutoTopicCreationOverride(topicName);

--- a/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java
+++ b/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java
@@ -2835,8 +2835,7 @@ public class BrokerService implements Closeable {
 
     public boolean isAllowAutoTopicCreation(final TopicName topicName) {
         //System topic can always be created automatically
-        if ((pulsar.getConfiguration().isSystemTopicEnabled() && isSystemTopic(topicName))
-            || NamespaceName.SYSTEM_NAMESPACE.equals(topicName.getNamespaceObject())) {
+        if ((pulsar.getConfiguration().isSystemTopicEnabled() && isSystemTopic(topicName))) {
             return true;
         }
         AutoTopicCreationOverride autoTopicCreationOverride = getAutoTopicCreationOverride(topicName);

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/systopic/PartitionedSystemTopicTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/systopic/PartitionedSystemTopicTest.java
@@ -112,7 +112,7 @@ public class PartitionedSystemTopicTest extends BrokerTestBase {
     }
 
     @Test
-    public void testProduceAndConsumeUnderSystemTopic() throws Exception {
+    public void testProduceAndConsumeUnderSystemNamespace() throws Exception {
         TenantInfo tenantInfo = TenantInfo
                 .builder()
                 .adminRoles(Sets.newHashSet("admin"))

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/systopic/PartitionedSystemTopicTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/systopic/PartitionedSystemTopicTest.java
@@ -19,21 +19,28 @@
 package org.apache.pulsar.broker.systopic;
 
 import com.google.common.collect.Sets;
+import lombok.Cleanup;
 import org.apache.commons.lang.RandomStringUtils;
 import org.apache.pulsar.broker.service.BrokerTestBase;
 import org.apache.pulsar.client.api.Consumer;
+import org.apache.pulsar.client.api.Message;
+import org.apache.pulsar.client.api.Producer;
+import org.apache.pulsar.client.api.SubscriptionInitialPosition;
+import org.apache.pulsar.client.api.SubscriptionType;
 import org.apache.pulsar.common.events.EventsTopicNames;
 import org.apache.pulsar.common.naming.NamespaceName;
+import org.apache.pulsar.common.policies.data.TenantInfo;
 import org.apache.pulsar.common.policies.data.TenantInfoImpl;
 import org.apache.pulsar.common.util.FutureUtil;
 import org.testng.Assert;
 import org.testng.annotations.AfterMethod;
 import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
-
+import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
 
 @Test(groups = "broker")
 public class PartitionedSystemTopicTest extends BrokerTestBase {
@@ -102,6 +109,30 @@ public class PartitionedSystemTopicTest extends BrokerTestBase {
         for (CompletableFuture<Consumer<byte[]>> consumer : futureList) {
             consumer.join().close();
         }
+    }
+
+    @Test
+    public void testProduceAndConsumeUnderSystemTopic() throws Exception {
+        TenantInfo tenantInfo = TenantInfo
+                .builder()
+                .adminRoles(Sets.newHashSet("admin"))
+                .allowedClusters(Sets.newHashSet("test"))
+                .build();
+        admin.tenants().createTenant("pulsar", tenantInfo);
+        admin.namespaces().createNamespace("pulsar/system", 2);
+        @Cleanup
+        Producer<byte[]> producer = pulsarClient.newProducer().topic("pulsar/system/__topic-1").create();
+        producer.send("test".getBytes(StandardCharsets.UTF_8));
+        @Cleanup
+        Consumer<byte[]> consumer = pulsarClient
+                .newConsumer()
+                .topic("pulsar/system/__topic-1")
+                .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
+                .subscriptionName("sub1")
+                .subscriptionType(SubscriptionType.Shared)
+                .subscribe();
+        Message<byte[]> receive = consumer.receive(5, TimeUnit.SECONDS);
+        Assert.assertNotNull(receive);
     }
 
 }

